### ISSUE TITLE
Question about the semantic label

### ISSUE THREAD

**Issue by Tianxinhuang** [wangyida/softpool#13]
Hello! Thank you for your insightful work! I've noticed that semantic labels seem to be adopted in the calculation of loss. So, is it needed to give a semantic label for each point with a network or something before training?

**Comment by wangyida** [comment]
Hi Tianxin,

Thanks for your interest on our work. Point-wise semantic labels are able to get used, but we didn't use it for ShapeNet completion in neither our paper nor complete3d benchmark. It merely influences the completion performance as far as we have tested. So you can just ignore it in train.py by not adding it into the geometric completion loss

Best,
Yida

**Comment by Tianxinhuang** [comment]
Yeah, so does it mean that we can just give a unrelated label while training because we don't have semantic label for our data? By the way, I'm very interested in the boundary loss mentioned in the paper, which can even improves simple PCN a lot. But I cannot find corresponding codes here. Could you please point it for me? Thank you very much!

**Comment by Tianxinhuang** [comment]
Yeah, so does it mean that we can just give a unrelated label while training when we don't have semantic label for our data? By the way, I'm very interested in the boundary loss mentioned in the paper, which can even improves simple PCN a lot. But I cannot find corresponding codes here. Could you please point it for me? Thank you very much!

**Comment by wangyida** [comment]
Yep, 'just give an unrelated label'. We have the *boundary loss* in the TensorFlow version, which matches our L_{boundary} in ECCV paper. We currently found that *expansion panalty* contributed in MSN (AAAI 20) performs slightly better than our L_{boundary}, so that the new parametric model in pytorch is trained with *expansion panalty* instead here https://github.com/wangyida/softpool/blob/master/pytorch/train.py#L84.

**Comment by Tianxinhuang** [comment]
OK, got it. Thank you very much.
